web/src/components/SubmissionsTable.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+  <v-data-table
+    class="submissions-table row-pointer"
+    :headers="headers"
+    :items="items"
+    :search="searchValue"
+    sort-by="similarity"
+    sort-desc
+    hide-default-footer
+    disable-pagination
+    must-sort
+    fixed-header
+    @click:row="rowClicked"
+  >
+    <template #item.name="{ item }">
+      <div class="submission-info">
+        <div class="submission-name">
+          {{ item.name }}
+        </div>
+
+        <div class="submission-path text--secondary">
+          <v-icon small>mdi-file-document-outline</v-icon>
+          <span>{{ item.path }}</span>
+        </div>
+      </div>
+    </template>
+
+    <template #item.label="{ item }">
+      <div class="submission-label">
+        <label-dot :label="item.label.label" :color="item.label.color" />
+        <label-text :label="item.label.label" colored />
+      </div>
+    </template>
+
+    <template #item.similarity="{ item }">
+      <span class="submission-similarity">
+        <similarity-display :similarity="item.similarity" progress dim-below-cutoff />
+      </span>

```

Comment:
An idea that popped in my mind, but I don't know how doable it is: showing which other submission(s) the current row has the highest similarity with (maybe only on hover?). What do you think?

Reply:
This is definitely possible but:
* This will be expensive, as all pairs matching a file must be found and sorted by similarity
* I would move this into a separate issue instead of patching it on this PR